### ISSUE TITLE
Security vulnerability in Pillow < 8.1.1

### ISSUE THREAD

**Issue by carlhiggs** [healthysustainablecities/global-indicators#116]
Hi all (esp. @shiqin-liu and @gboeing !)

I just received a security warning for the Docker requirement to use Pillow 8.0.*; apparently for versions less than 8.1.1 there's  a vulnerability in the regex code that can be exploited by a malicious pdf.  

I don't see it being an issue for the project per se, except that its a bad look having a dependabot warning on the repo page 

I was just making those few changes to the setup_config.py for the other issue (change pre-processed study region folder from 'input' to 'study_region')  --- so i just tested a rebuild of the image making the suggested requirement change
`Pillow>=8.1.1`

It appears to work fine for me, and I expect it will address the warning (I'm not sure what Pillow is used for exactly in the project; I expect its a dependency of one of our packages).  

Are you happy for me to push this change up in both code (requirements.txt, and environment.yml) and to DockerHub?

**Comment by carlhiggs** [comment]
I committed changes for this earlier today, as per above.  In advertantly, as per message in https://github.com/global-healthy-liveable-cities/global-indicators/pull/115 i did this to the main branch instead of a working one I had created.  

So, I think this means it has been addressed in the repository  --- but I have not yet pushed the updated image up to DockerHub, so I won't yet close this issue.  I will be away for the next few days, but can push the image up to DockerHub when I return once confirmed its a good idea, or happy for others to re-create it and do this.  Or, perhaps there is another approach.

So i'll leave this issue open for now.

**Comment by shiqin-liu** [comment]
I did not realize this warning on my end, but it seems harmless? I am fine with this change if all agree :)

**Comment by gboeing** [comment]
Fleshing this out with some more details. The change @carlhiggs referenced was made in bab2254d1abfc02bc1f3f60eb322ccfbb8271ab1. The `pillow` vulnerability is documented [here](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-25292) and the fix is in their change log [here](https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html).

Currently our proposed [fix](https://github.com/global-healthy-liveable-cities/global-indicators/commit/bab2254d1abfc02bc1f3f60eb322ccfbb8271ab1#diff-43595f15069f96010c3ba429e9aa268aecf572c2a9963dde285cad2c0ad3839bR19) pins the version hard to patch `8.1.1`. Pinning all the way down to a patch can make it impossible to rebuild an environment in the future. I would advise pinning to the minor version instead, for example, `8.1.*` or even `8.2.*` if our code is compatible with it.

**Comment by carlhiggs** [comment]
Hey, i've just created a pull request for an update on the 'fix' branch where I amended the requirements text to have `pillow >= 8.1.1`; things appeared to build and work successfully.  Re-reading, I just realised I could have changed this to fix to minor version 8.2.* as per suggestion --- let me know if that's preferable and I'll be happy to modify and push up an updated version.  Otherwise, if you're happy with the pull request we could merge it in and close this issue!

**Comment by gboeing** [comment]
Yes let's pin it to 8.2.*

Otherwise, if we rebuild a year from now, it may install v8.4.0 (for example), which our own code may end up having minor incompatibilities with. If our code is tested and working with 8.2.*, then let's pin to that minor version.

**Comment by carlhiggs** [comment]
Thanks Geoff; I attempted setting pillow requirements to 8.2.*, however this cascaded to a large number of package version conflicts when attempting to build the image.

A subsequent build using the more restricted range of `pillow >= 8.1.1,8.2.0` was successful, and so I have just committed this change (updates to requirements.txt and environment.yml).

Let me know if you think this works, and if so I can push the image changes up to Dockerhub and merge in the pull request from the 'fix' branch.
